metricbeat/module/couchdb/server/server.go

```diff
@@ -76,8 +95,76 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 		return errors.Wrap(err, "error in http fetch")
 	}
 
-	event, _ := eventMapping(content)
-	reporter.Event(mb.Event{MetricSetFields: event})
+	if m.fetcher == nil {
```

Comment:
nit: I would hide this condition inside retrieveFetcher. The provider creates an instance if it's not ready.

Reply:
Moved it inside, thanks!